internal/generate/crd/crd.go

```diff
@@ -0,0 +1,348 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	genutil "github.com/operator-framework/operator-sdk/internal/generate/util"
+	"github.com/operator-framework/operator-sdk/internal/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/util/fileutil"
+	"github.com/operator-framework/operator-sdk/internal/util/k8sutil"
+	"github.com/operator-framework/operator-sdk/internal/util/yamlutil"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	apiextv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// crdGenerator configures the CustomResourceDefintion manifest generator
+// for Go and non-Go projects.
+type crdGenerator struct {
+	// Dir containing relevant input files.
+	inputDir string
+	// Dir in which to generate output files.
+	outputDir string
+	// isOperatorGo is true when the operator is written in Go.
+	isOperatorGo bool
+	// resource contains API information used to configure single-CRD generation.
+	// This is only required when isOperatorGo is false.
+	resource scaffold.Resource
```

Comment:
```suggestion
	resource *scaffold.Resource
```

Reply:
I'd rather not persist heap-allocated objects when we don't have to.